src/main/scala/com/datastax/gatling/plugin/utils/CqlPreparedStatementUtil.scala

```diff
@@ -44,12 +45,80 @@ trait CqlPreparedStatementUtil {
   def getParamsList(preparedStatement: PreparedStatement): List[Int]
 }
 
+object SessionCollectionResolver {
+  def get(session:Session, name:String):Option[Any] = session.attributes.get(name)
+
+  def getClz[T <: Any](session:Session, name:String):Option[Class[T]] = {
+    get(session,name).flatMap((sessionVal) => {
+      sessionVal match {
+        case clz:Class[T] => Option(clz)
+        case _ => Option.empty
+      }
+    })
+  }
+
+  def getIterable[T <: Any](session:Session, name:String):Option[lang.Iterable[T]] = {
+    get(session,name).flatMap((sessionVal) => {
+      sessionVal match {
+        case rv:Iterable[T] => Option(rv.asJava)
+        case rv:lang.Iterable[T] => Option(rv)
+        case _ => Option.empty
+      }
+    })
+  }
+
+  def getMap[K <: Any, V <: Any](session:Session, name:String):Option[util.Map[K,V]] = {
+    get(session,name).flatMap((sessionVal) => {
+      sessionVal match {
+        case rv:Map[K,V] => Option(rv.asJava)
+        case rv:util.Map[K,V] => Option(rv)
+        case _ => Option.empty
+      }
+    })
+  }
+
+  def getIterableClz[T <: Any](session:Session, name:String):Class[_ <: T] = {
+    val sessionOption:Option[Class[T]] = getClz(session, name + "-clz")
+    if (sessionOption.isDefined) {
+      sessionOption.get
+    } else {
+      val iterableOption:Option[lang.Iterable[T]] = getIterable(session, name)
+      if (iterableOption.isEmpty) {
+        throw new IllegalStateException("Iterable element class wasn't defined in Gatling session and Iterable is unavailable, cannot determine list type")
+      }
+      val iterator = iterableOption.get.iterator
+      if (!iterator.hasNext) {
+        throw new IllegalStateException("Iterable element class wasn't defined in Gatling session and Iterable is empty, cannot determine list type")
+      }
+      iterator.next.getClass
+    }
+  }
+
+  def getMapClzs[K <: Any, V <: Any](session:Session, name:String):(Class[_ <: K],Class[_ <: V]) = {
```

Comment:
After my suggestions for other methods, this could be simplified to:
```
  def getMapClzs[K <: Any, V <: Any](session:Session, name:String):(Class[_ <: K],Class[_ <: V]) = {
    val sessionKeyOption:Option[Class[K]] = getClz(session, name + "-key-clz")
    val sessionValOption:Option[Class[V]] = getClz(session, name + "-val-clz")
    if (sessionKeyOption.isDefined && sessionValOption.isDefined) {
      (sessionKeyOption.get, sessionValOption.get)
    } else {
      val iterator = getMap(session, name).iterator
      if (!iterator.hasNext) {
        throw new IllegalStateException("Map classes wasn't defined in Gatling session and Map is empty, cannot determine list type")
      }
      val entry = iterator.next
      (entry._1.getClass, entry._2.getClass)
    }
  }
```